src/aerospike_clj/key.clj

```diff
@@ -0,0 +1,37 @@
+(ns aerospike-clj.key
+  (:import [com.aerospike.client Key Value]
+           [com.aerospike.client.util ThreadLocalData]))
+
+(defprotocol AerospikeIndex
+  (create-key-method ^Key [this ^String as-namesapce ^String set-name]))
+
+(extend-protocol AerospikeIndex
+  (Class/forName "[B") 
+  (create-key-method ^Key [this ^String as-namesapce ^String set-name]
+    (when (< (ThreadLocalData/DefaultBufferSize) (+ 1 (alength #^bytes this) (.length set-name)))
```

Comment:
Why not `<=` and lose the `(+ 1 `?

Reply:
this is how it was defined in the Java code. I'll change that.